TYPO3.Neos/Resources/Public/JavaScript/Content/Components/ContentContextBar.js

```diff
@@ -2,38 +2,55 @@
  * Context context bar
  */
 define(
-[
-	'emberjs',
-	'./ContextBar',
-	'./ContentDimensionSelector',
-	'../FullScreenController',
-	'../Application',
-	'Shared/Configuration',
-	'text!./ContentContextBar.html'
-], function(
-	Ember,
-	ContextBar,
-	ContentDimensionSelector,
-	FullScreenController,
-	ContentModule,
-	Configuration,
-	template
-) {
-	return ContextBar.extend({
-		classNames: 'neos-content-context-bar',
-		template: Ember.Handlebars.compile(template),
-		ContentDimensionSelector: ContentDimensionSelector,
-		fullScreenController: FullScreenController,
-		Configuration: Configuration,
-		init: function() {
-			this.updateCurrentUri();
-			var that = this;
-			ContentModule.on('pageLoaded', function() {
-				that.updateCurrentUri();
-			});
-		},
-		updateCurrentUri: function() {
-			this.set('liveUri', location.href.replace(/@[A-Za-z0-9;&,\-_=]+/g, ''));
-		}
-	});
-});
\ No newline at end of file
+  [
+    'emberjs',
+    './ContextBar',
+    './ContentDimensionSelector',
+    './ContentDimensionController',
+    '../FullScreenController',
+    '../Application',
+    './TargetWorkspaceController',
+    'Shared/Configuration',
+    'text!./ContentContextBar.html'
+  ], function (Ember,
+               ContextBar,
+               ContentDimensionSelector,
+               ContentDimensionController,
+               FullScreenController,
+               ContentModule,
+               TargetWorkspaceController,
+               Configuration,
+               template) {
+    return ContextBar.extend({
+      classNames: 'neos-content-context-bar',
+      template: Ember.Handlebars.compile(template),
+      ContentDimensionSelector: ContentDimensionSelector,
+      contentDimensionController: ContentDimensionController,
+      fullScreenController: FullScreenController,
+      targetWorkspaceController: TargetWorkspaceController,
+      Configuration: Configuration,
+
+      /**
+       * Update the preview uri when the page is loaded (happens when navigating, switching
+       * target workspace or changing content dimensions).
+       */
+      init: function () {
+        var that = this;
+        ContentModule.on('pageLoaded', function () {
+          that.updatePreviewUri();
+        });
```

Comment:
Indeed. Well, even an initial page load is a page load, so why doesn't it say pageLoaded? pfft.